src/api/app/models/configuration.rb

```diff
@@ -36,18 +48,17 @@ def method_missing(method_name, ...)
       end
     end
 
-    # overwrite update function as the one in active record expects an id
-    def update(opts)
-      Configuration.first.update(opts)
+    def ldap_enabled?
+      CONFIG['ldap_mode'] == :on
     end
 
     # Check if ldap group support is enabled?
     def ldapgroup_enabled?
       CONFIG['ldap_mode'] == :on && CONFIG['ldap_group_support'] == :on
     end
 
-    def ldap_enabled?
-      CONFIG['ldap_mode'] == :on
+    def passwords_changable?(user = nil)
```

Comment:
But it was not a class method before. I thought you where limiting class methods to things that don't need the database? I did I misunderstand the purpose of this?

Reply:
> But it was not a class method before. I thought you where limiting class methods to things that don't need the database? I did I misunderstand the purpose of this?

The purpose of this refactoring is to move as many methods as possible to prevent from calling `first.send(...)` when we don't need it. That's because the method `first` queries the database. More explicitly, I'm talking about this line: https://github.com/openSUSE/open-build-service/blob/a875cf629f93ea1c569782b7bf9a074daa4d0245/src/api/app/models/configuration.rb#L33

Until now, all methods could be moved to the class section because they don't make use of the singleton record.

The case of the `passwords_changable?` method is special because it reads the `change_password` column of the singleton record. Moving `passwords_changable?` to the class method section has the following benefits:
- we prevent from calling `first.send(...)`, so we don't query the database, and
- we can benefit from using `change_password` as a class method (also through `method_missing`), which will make use of the cached record. See: https://github.com/openSUSE/open-build-service/blob/a875cf629f93ea1c569782b7bf9a074daa4d0245/src/api/app/models/configuration.rb#L31

That's why I also moved this method up to the class section. I don't have in mind to move any other method.

Reply:
Hm I don't think there is any difference between doing a database query from `change_password` or from `fetch.send`. I wouldn't move it to be a class method, it's confusing.

Reply:
> Hm I don't think there is any difference between doing a database query from `change_password` or from `fetch.send`. I wouldn't move it to be a class method, it's confusing.

`fetch.send` will not perform a database query. I will use the cache, most of the times...